packages/fluentui/docs/src/pages/Debugging.mdx

```diff
@@ -0,0 +1,121 @@
+export const meta = {
+  title: 'Debugging',
+  previous: { name: 'Performance', url: 'performance' },
+};
+
+## Overview
+
+Fluent UI provides tools for developers to debug styles and measure performance.
+
+## Debugging styles
+
+### How merging works
+
+As it was described in [Theming](/theming), a theme consists of three different building blocks:
+
+1. Site variables,
+2. Component variables,
+3. Component styles (for individual component slots).
+
+Themes can be nested, component variables and styles can be overriden in a component instance:
+
+```jsx
+<Provider theme={outerTheme}>
+    <Provider theme={innerTheme}>
+        <Component variables={componentVariablesOverrides} design={designStyleOverrides} styles={componentStyleOverrides} />
+    </Provider>
+</Provider>
+```
+
+In order to compute CSS styles, all the theme pieces are merged:
+
+![Merge order](public/images/merge-themes.png)
+
+1. First, site variables are merged together,
+2. Merged site variables are passed as input to component variables, resolved component variables are merged together,
+3. Merged component variables (as well as merged site variables) are passed as input to component styles, resolved component styles are merged together,
+4. Merges component styles for each component slot are passed to CSSInJS renderer to process the styles with plugins (RTL, expand, vendor prefixing), generate CSS classes and add the styles to DOM.
+
+As the merged component styles are processes by CSSInJS and the way how those are added to DOM is CSSInJS specific (Fela uses atomic CSS classes adding a class for each CSS attribute) it might be difficult to debug the styles using browser's development tools.
```

Comment:
```suggestion
As the merged component styles are processed by CSSInJS and the way how those are added to DOM is CSSInJS specific (Fela uses atomic CSS classes adding a class for each CSS attribute) it might be difficult to debug the styles using browser's development tools.
```